layers/core_checks/cc_render_pass.cpp

```diff
@@ -4003,28 +4008,91 @@ bool CoreChecks::OutsideRenderPass(const vvl::CommandBuffer &cb_state, const Loc
     if (!cb_state) return false;
     bool skip = false;
     skip |= ValidateCmd(*cb_state, error_obj.location);
     if (skip) return skip;  // basic validation failed, might have null pointers
     ASSERT_AND_RETURN_SKIP(cb_state->active_render_pass);
 
+    const bool is_ext = error_obj.location.function == Func::vkCmdEndRendering2EXT;
+
     if (!cb_state->active_render_pass->UsesDynamicRendering()) {
-        skip |= LogError("VUID-vkCmdEndRendering-None-06161", commandBuffer, error_obj.location,
+        const char *vuid = is_ext ? "VUID-vkCmdEndRendering2EXT-None-10610" : "VUID-vkCmdEndRendering-None-06161";
+        skip |= LogError(vuid, commandBuffer, error_obj.location,
                          "in a render pass instance that was not begun with vkCmdBeginRendering().");
     }
     if (cb_state->active_render_pass->use_dynamic_rendering_inherited) {
-        skip |= LogError("VUID-vkCmdEndRendering-commandBuffer-06162", commandBuffer, error_obj.location,
+        const char *vuid = is_ext ? "VUID-vkCmdEndRendering2EXT-commandBuffer-10611" : "VUID-vkCmdEndRendering-commandBuffer-06162";
+        skip |= LogError(vuid, commandBuffer, error_obj.location,
+                         "in a render pass instance that was not begun in this command buffer.");
+    }
+    if (cb_state->transform_feedback_active) {
+        const char *vuid = is_ext ? "VUID-vkCmdEndRendering2EXT-None-10612" : "VUID-vkCmdEndRendering-None-06781";
+        skip |= LogError(vuid, commandBuffer, error_obj.location,
                          "in a render pass instance that was not begun in this command buffer.");
     }
     for (const auto &query : cb_state->render_pass_queries) {
         const LogObjectList objlist(commandBuffer, query.pool);
-        skip |= LogError("VUID-vkCmdEndRendering-None-06999", objlist, error_obj.location,
-                         "query %" PRIu32 " from %s was began in the render pass, but never ended.", query.slot,
-                         FormatHandle(query.pool).c_str());
+        const char *vuid = is_ext ? "VUID-vkCmdEndRendering2EXT-None-10613" : "VUID-vkCmdEndRendering-None-06999";
+        skip |=
+            LogError(vuid, objlist, error_obj.location, "query %" PRIu32 " from %s was began in the render pass, but never ended.",
+                     query.slot, FormatHandle(query.pool).c_str());
     }
+
+    return skip;
+}
+
+bool CoreChecks::PreCallValidateCmdEndRendering(VkCommandBuffer commandBuffer, const ErrorObject &error_obj) const {
+    return ValidateCmdEndRendering(commandBuffer, error_obj);
+}
+
+bool CoreChecks::PreCallValidateCmdEndRendering2EXT(VkCommandBuffer commandBuffer, const VkRenderingEndInfoEXT *pRenderingEndInfo,
+                                                    const ErrorObject &error_obj) const {
+    bool skip = false;
+
+    skip |= ValidateCmdEndRendering(commandBuffer, error_obj);
+
+    auto cb_state = GetRead<vvl::CommandBuffer>(commandBuffer);
+    if (!cb_state) return false;
```

Comment:
```suggestion
    ASSERT_AND_RETURN_SKIP(cb_state)
```

any `vkCmd` call needs to have a valid cb_state, or there is no way the loader would have found the layer in the first place